Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -256,16 +286,25 @@ private WeaveTypeResult WeaveGeneratedType(TypeDefinition type)
             var interfaceType = _moduleDefinition.GetType($"{@namespace}.Generated", interfaceName);
 
             var persistedProperties = new List<WeavePropertyResult>();
+            var backingFields = new HashSet<MetadataToken>();
 
             // We need to weave all (and only) the properties in the accessor interface
             foreach (var interfaceProperty in interfaceType.Properties)
             {
                 var prop = type.Properties.First(p => p.Name == interfaceProperty.Name);
-
                 try
                 {
+                    // Stash and remove the backing field before weaving as it depends on get method.
+                    var backingField = prop.GetBackingField();
+                    if (backingField != null) {
+                        type.Fields.Remove(backingField.Resolve());   
```

Comment:
I am removing it at this point as it's possible there won't be any constructors referencing an existing backing field, right? This seemed like the best place to remove it, I can also move it to a function so it's cleaner